engine/src/witness/common/cf_parameters.rs

```diff
@@ -0,0 +1,54 @@
+use codec::{Decode, Encode, MaxEncodedLen};
+use scale_info::TypeInfo;
+use sp_core::ConstU32;
+
+use anyhow::{anyhow, Result};
+use cf_chains::{CcmAdditionalData, ChannelRefundParameters, MAX_CCM_ADDITIONAL_DATA_LENGTH};
+use cf_primitives::{BasisPoints, DcaParameters};
+use frame_support::sp_runtime::BoundedVec;
+
+pub const MAX_VAULT_SWAP_PARAMETERS_LENGTH: u32 = 1_000;
+pub const MAX_CF_PARAM_LENGTH: u32 =
+	MAX_CCM_ADDITIONAL_DATA_LENGTH + MAX_VAULT_SWAP_PARAMETERS_LENGTH;
+pub type CfParameters = BoundedVec<u8, ConstU32<MAX_CF_PARAM_LENGTH>>;
+
+#[derive(Encode, Decode, MaxEncodedLen, TypeInfo, Clone, PartialEq, Debug)]
+pub struct VaultCfParameters {
+	pub ccm_additional_data: Option<CcmAdditionalData>,
+	pub vault_swap_parameters: VaultSwapParameters,
+}
```

Comment:
That's also what I was initially going for but I thought that if in the future we add more features it makes more sense to keep adding them as another additional parameter in the `VaultCfParameters`. And in that case it will end up with those features split by the `ccm_additional_data`. It seemed easier to have first the ccm data and then the rest. Not a strong opinion though.